InformationScripting/src/parsing/QueryParser.h

```diff
@@ -60,10 +61,10 @@ class INFORMATIONSCRIPTING_API QueryParser
 		Type typeOf(const QString& text);
 		QPair<QStringList, QList<QChar> > split(const QString& text, const QList<QChar>& splitChars);
 
-		Query* parseQuery(const QString& text);
-		QList<Query*> parseList(const QString& text);
-		Query* parseOperator(const QString& text, bool connectInput = false);
-		QList<Query*> parseOperatorPart(const QString& text);
+		Query* parseQuery(Query* parent, const QString& text);
```

Comment:
Hm If we imagine someone implementing a new Query which needs access to the QE it might make sense if it is already there? On the other hand I agree that most queries don't need to know about it. Overall I slightly tend to have it in the Query base class, in a way the queries still don't need to care about it, if they don't want to use it they don't set the field.